eodatasets3/model.py

```diff
@@ -373,6 +372,81 @@ def producer_abbreviated(self) -> Optional[str]:
             )
 
 
+class ComplicatedNamingConventionsAlchemist(ComplicatedNamingConventions):
+
+    """
+    This class is inherited from ComplicatedNamingConventions
+    and overrides few attributes specific to C3 dataprocessing at the Alchemist end.
+    """
+
+    @classmethod
+    def for_c3_processing(cls, dataset: EoFields, uri=DEA_URI_PREFIX):
+        """
+        The required fields for the c3 data processing are controlled here.
+        """
+        return cls(
+            dataset=dataset,
+            base_product_uri=uri,
+            required_fields=(
+                "eo:platform",
+                "odc:dataset_version",
+                "odc:collection_number",
+                "odc:processing_datetime",
+                "odc:producer",
+                "odc:product_family",
+                "odc:region_code",
+                "dea:dataset_maturity",
+            ),
+        )
+
+    @property
+    def _org_collection_number(self) -> Optional[int]:
+
+        # Deliberately fail if collection_number is not defined in the config yaml
+        return int(self.dataset.collection_number)
+
+    def _product_group(self, subname=None) -> str:
```

Comment:
These product groups are supposed to have the collection number in them... I'm guessing that comes in somewhere else... but perhaps it shouldn't, and it should come in here?

Reply:
For the earlier logic, collection_number was simply a part of `dataset_version`. This feature requires two different things from which `collection_number` cannot be parsed from `dataset_version`